pages/[p].tsx

```diff
@@ -35,30 +33,59 @@ export default function Donate({
         const currency = getStoredCurrency();
         setInternalCurrencyCode(currency);
         setCurrencyCode(currency);
-        const project = await getRequest(`/app/projects/${router.query.p}?_scope=extended&currency=${currency}`);
-        setProject(project);
-        setShowSingleProject(true);
+        const result = await getRequestWithLocale(
+          `/app/projects/${router.query.p}?_scope=extended&currency=${currency}`
+        );
+        setProject(result);
       }
     }
-    if (router.query.p) {
-      loadProject();
-    }
-  }, [router.query.p, currencyCode]);
+    loadProject();
+  }, [currencyCode]);
 
   const ProjectProps = {
     project,
+    currencyCode,
+    setCurrencyCode,
   };
 
   return (
     <>
-      {project ? <GetProjectMeta {...ProjectProps} /> : null}
-      {initialized ? (
-        project && initialized ? (
-          <SingleProjectDetails {...ProjectProps} />
-        ) : (
-          <></>
-        )
-      ) : null}
+      {project && (
+        <>
+          <GetProjectMeta {...ProjectProps} />
+          <MapLayout {...ProjectProps} />
+          {initialized ? <SingleProjectDetails {...ProjectProps} /> : null}
+          <Credits setCurrencyCode={setCurrencyCode} />
+        </>
+      )}
     </>
   );
 }
+
+export const getStaticProps: GetStaticProps = async (context: any) => {
+  let projectData = await getRequest(
+    `/app/projects/${context.params.p}?_scope=extended`
+  );
+
+  return {
+    props: {
+      projectData,
+    },
+    // Next.js will attempt to re-generate the page:
+    // - When a request comes in
+    // - At most once every 300 seconds
+    revalidate: 300, // In seconds
+  };
+};
+
+export const getStaticPaths: GetStaticPaths = async () => {
+  let paths: { params: { p: any } }[] = [];
+  let projects = await getRequest(`/app/projects`);
+  projects.forEach((project: { slug: any }) => {
+    paths.push({ params: { p: project.slug } });
+  });
+  return {
+    paths: paths,
+    fallback: false,
+  };
+};
```

Comment:
I will have to study the documentation how this concept of _automatically generated pages as static HTML + JSON_ works, but seeing this at deployment time I wonder if newly added projects will be available and changed or deleted projects will updated or disabled only if we do another deployment of our website?

```
19:40:46.328 | Page                                                           Size     First Load JS
-- | --
19:40:46.329 | ┌ ● / (ISR: 300 Seconds)                                       282 kB          1.2 MB
19:40:46.329 | ├   └ css/c8c6da4b7a736198b400.css                             5.69 kB
19:40:46.329 | ├   /_app                                                      0 B             232 kB
19:40:46.329 | ├ ● /[p]                                                       316 kB         1.22 MB
19:40:46.329 | ├   └ css/a7c63ce8f0b190f61a01.css                             6.4 kB
19:40:46.329 | ├   ├ /conserve-natural-forests
19:40:46.329 | ├   ├ /savingnaturebrazil
19:40:46.329 | ├   ├ /bayka-com-ar
19:40:46.329 | ├   └ [+145 more paths]
...
19:40:46.332 | ●  (SSG)     automatically generated as static HTML + JSON (uses getStaticProps)
-- | --

```

Reply:
any path not generated at build time will get generated at runtime if `fallback: true`
https://nextjs.org/docs/basic-features/data-fetching#fallback-true